configure.ac

```diff
@@ -132,6 +132,11 @@ if test "$enable_ssh" != "no"; then
     AC_DEFINE_UNQUOTED(HAVE_SSH_GET_SERVER_PUBLICKEY, 1, Whether ssh_get_server_publickey is available)
   ])
 
+  AC_CHECK_LIB(ssh, ssh_session_has_known_hosts_entry, [
+      AC_DEFINE_UNQUOTED(HAVE_SSH_SESSION_HAS_KNOWN_HOSTS_ENTRY, 1, Whether ssh_has_known_host_entry is available)
+    has_session_known_host_entry="yes"
```

Comment:
OK if that works, but it looks rather unusual, and the indentation is off, too. Why isn't the `AM_CONDITIONAL` below enough to define (or not) `HAVE_SSH_SESSION_HAS_KNOWN_HOSTS_ENTRY`?

Also, please have a look at my PR #10241. `AC_CHECK_DECLS` is much simpler to use.

Reply:
Ah thanks, I'll take a look. To be honest I don't have that much experience with automake.

Reply:
Also I need both a  preprocessor variable and a variable inside the Makefile. Can that be achieved with just `AM_CONDITIONAL`?

Reply:
`AC_CHECK_DECLS` automatically defines an `ac_cv_have_decl_ssh_session_has_known_hosts_entry` shell variable in this case, which you can use with `AM_CONDITIONAL` afterwards.